src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/UIUtility.cs

```diff
@@ -64,15 +64,17 @@ public static string NumberToString(long number, IFormatProvider culture)
 
             while (v >= 1000)
             {
-                v /= 1000;
+                v = Math.Round(v / 1000, 2);
                 ++exp;
             }
 
+            string format = v < 999 ? "{0:G3}{1}" : "{0:G4}{1}"; // If number is 999561 then v = 999.56 so with G3 which result in 1000k ~ 1E+03K, but with G4 it result in 999.6K.
```

Comment:
I added logic for handling v >= 999, we handle with 1 more precision , but I didn't for v< 999 otherwise it's kind of breaking experience for many users seeing too much detailed information.

Reply:
Isn't showing a 4th digit of precision when all other cases only show 3 also a kind of "breaking" experience?

How about `if (v > 999.0) { v = 999.0; }`?

Reply:
@zivkan 
Ok. Taken your suggestion. Please check now.